tests/ClientResolverTest.php

```diff
@@ -1519,4 +1520,199 @@ public function testResolveFromEnvIniUseAwsSharedFiles(): void
             }
         }
     }
+
+    /**
+     * @param bool $isExpected
+     * @param array $expectedValue
+     * @param array $args
+     * @param string|null $ini
+     * @param string|null $env
+     * @return void
+     *
+     * @dataProvider resolvesAuthSchemePreferenceProvider
+     */
+    public function testResolvesAuthSchemePreference(
+        bool $isExpected,
+        array $expectedValue,
+        array $args = [],
+        ?string $ini = null,
+        ?string $env = null,
+    ): void
+    {
+        // Callback list to be executed after test is executed
+        $onCompletionCallbacks = [];
+        if ($ini !== null) {
+            $currentEnvAwsConfigFile = getenv('AWS_CONFIG_FILE');
+            $tempDir = sys_get_temp_dir() . '/test-auth-preference/.aws/';
+            $tempConfigFile = $tempDir . 'config';
+            $configContent = "[default]\nauth_scheme_preference=$ini";
+
+            // If the temp directory does not exist then we create it
+            if (!is_dir($tempDir)) {
+                mkdir($tempDir, 0777, true);
+            }
+
+            file_put_contents($tempConfigFile, $configContent);
+            putenv("AWS_CONFIG_FILE=$tempConfigFile");
+
+            $onCompletionCallbacks[] = function () use (
+                $tempConfigFile,
+                $tempDir,
+                $currentEnvAwsConfigFile
+            ) {
+                if ($currentEnvAwsConfigFile !== false) {
+                    putenv("AWS_CONFIG_FILE=$tempConfigFile");
```

Comment:
This still sets the original config file to the wrong value

Reply:
Could you please provide more details for why this will set it to the wrong value?
In line 1545 I try to get the current value for it, if not value is set then getenv will return false, which is the why of my condition here, otherwise, if it has a value then I will set it back.

Reply:
`$tempConfigFile = $tempDir . 'config';`

is what `AWS_CONFIG_FILE` is being set back to.  Should it not be $`currentEnvAwsConfigFile`?